ranger/config/commands.py

```diff
@@ -1990,3 +1990,34 @@ def make_safe_path(dst):
 
     def execute(self):
         return self.fm.paste(make_safe_path=paste_ext.make_safe_path)
+
+
+class confirm(Command):
+    """
+    :confirm <command> [ARGS...]
+
+    Can be used with any command. Ranger will ask if it should execute the
+    given command.
+    """
+
+    def execute(self):
+        if len(self.args) <= 1:
+            self.fm.notify("Nothing to confirm.")
+        elif self.arg(1) == 'chain':
+            # TODO: This doesn't work for chained chain commands.
+            for command in [arg.lstrip() for arg in self.rest(2).split(';')]:
```

Comment:
Will this ask for each `:chain` subcommand separately? I think it would make more sense to ask once for the whole chain. What was the rationale behind this decision?

Reply:
I think this stems from a misunderstanding of something I said. If we do listing of the files a command operates on then we'd have to recurse for chain. But the PR doesn't do that yet so all it would do is make the commands slightly more readable (because they're shorter) when confirming at the cost of having to confirm each command in a chain. But in addition the way the chain is split doesn't work, cf. the TODO I added.